### PR TITLE
Pipe plugins throwing error will bypass triggerPipes error management

### PR DESCRIPTION
**Pull request by scottinet**
# Description

Whenever a pipe function throws an error instead of returning a rejected promise, the following happens:
* the error is embedded in a `PluginImplementationError` error, without first checking if the rejected error is a `KuzzleError` (the pipe function may throw a KuzzleError-like error on purpose for instance)
* the pipe timers aren't stopped, leading to erroneous timeout logs being written
* the error management in the `triggerPipes` function is bypassed

This PR fixes how the caught error is handled by the pipe wrapper.

# How to reproduce

Take this perfectly valid pipe function, and assume that it is plugged to the `document:beforeCreate` event:

```js
function validateSomeDocuments(request) {
  if (/* some unmet condition */) {
    throw new this.context.errors.BadRequestError('unmet conditions error');
  }

  return Promise.resolve(request);
}
```

Whenever a document with unmet conditions is submitted, erroneous timeout logs appear in Kuzzle's log files, and users get a `PluginImplementationError` response instead of a `BadRequestError` one

# Boyscout

Cleaner require instructions using destructuring assignment

### CODE REVIEW

**Comment by kuzzle** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![INFO][INFO] [pluginsManager.js#L729](https://github.com/kuzzleio/kuzzle/blob/76eb55f75f5ebea73ff8269a8c271a6c277b3542/lib/api/core/plugins/pluginsManager.js#L729): Complete the task associated to this TODO comment. [rule](http://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'

**Comment by kuzzle** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![INFO][INFO] [pluginsManager.js#L729](https://github.com/kuzzleio/kuzzle/blob/d3a1eab0c06315a26afd28601704e2f92de709d3/lib/api/core/plugins/pluginsManager.js#L729): Complete the task associated to this TODO comment. [rule](http://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'

File: lib/api/core/plugins/pluginsManager.js [hunk 1]
```diff
@@ -418,11 +421,16 @@ class PluginsManager {
...
         }
       }
       catch (error) {
-        throw new PluginImplementationError(error);
+        if (error instanceof KuzzleError) {
+          return cb(error);
+        }
+
+        cb(new PluginImplementationError(
+          `Plugin ${name} pipe for event '${event}' throwed a non-Kuzzle error: ${error}`));
```
Comment (benoitvidis) [inline lib/api/core/plugins/pluginsManager.js]:
```suggestion
          `Plugin ${name} pipe for event '${event}' threw a non-Kuzzle error: ${error}`));
```

**Comment by kuzzle** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![INFO][INFO] [pluginsManager.js#L729](https://github.com/kuzzleio/kuzzle/blob/d30dc0900a37dfce1426cd2e0aef6b6a879bf429/lib/api/core/plugins/pluginsManager.js#L729): Complete the task associated to this TODO comment. [rule](http://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'

**Comment by kuzzle** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![INFO][INFO] [pluginsManager.js#L729](https://github.com/kuzzleio/kuzzle/blob/205740f3f4d181b2088c35f25b50bdfb7b9c91d6/lib/api/core/plugins/pluginsManager.js#L729): Complete the task associated to this TODO comment. [rule](http://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'

**Pull request merged**